File: openspec-py/app/repository/in_memory_employee_repository.py
from typing import Optional

from app.model.employee import Employee
from app.repository.employee_repository import EmployeeRepository


class InMemoryEmployeeRepository(EmployeeRepository):
    def __init__(self) -> None:
        self._store: dict[int, Employee] = {}
        self._sequence = 1

    def find_all(self) -> list[Employee]:
        return list(self._store.values())

    def find_by_id(self, id: int) -> Optional[Employee]:
        return self._store.get(id)

    def save(self, employee: Employee) -> Employee:
        if employee.id is None:
            employee.id = self._sequence
            self._sequence += 1
        self._store[employee.id] = employee
        return employee

    def delete_by_id(self, id: int) -> None:
        self._store.pop(id, None)

    def exists_by_id(self, id: int) -> bool:
        return id in self._store

File: openspec-php/src/Repository/InMemoryEmployeeRepository.php
<?php

declare(strict_types=1);

namespace App\Repository;

use App\Model\Employee;

class InMemoryEmployeeRepository implements EmployeeRepositoryInterface
{
    /** @var array<int, Employee> */
    private array $store = [];

    private int $sequence = 1;

    public function findAll(): array
    {
        return array_values($this->store);
    }

    public function findById(int $id): ?Employee
    {
        return $this->store[$id] ?? null;
    }

    public function save(Employee $employee): Employee
    {
        if ($employee->getId() === null) {
            $employee->setId($this->sequence++);
        }
        $this->store[$employee->getId()] = $employee;

        return $employee;
    }

    public function deleteById(int $id): void
    {
        unset($this->store[$id]);
    }

    public function existsById(int $id): bool
    {
        return isset($this->store[$id]);
    }
}

